### ISSUE TITLE
[IDEA] Use Prettier

### ISSUE THREAD

**Issue by NicolasPetton** [Jermolene/TiddlyWiki5#4625]
**Is your feature request related to a problem? Please describe.**

The formatting of JavaScript in TIddlyWiki is rather unconventional, and discussing formatting within a community is always a headache.

I think the current formatting can be quite off-putting and as a consequence be a barrier to contributions (it certainly was for me when I first looked at the code of TW).

**Describe the solution you'd like**

Prettier (https://prettier.io/) is a formatting tool that has been widely adopted by OSS projects.
The main advantage I see it that there's no bike-shedding when using prettier, as it has (on purpose) almost no options.

The formatted code also follows widespread conventions, it has plugins for most text editors, can be applied automatically on a codebase, and can be setup as a CI job.

**What would TW JS code look like?**

Today TW code looks like this:

```js
var Commander = function(commandTokens,callback,wiki,streams) {
	var path = require("path");
	this.commandTokens = commandTokens;
	this.nextToken = 0;
	this.callback = callback;
	this.wiki = wiki;
	this.streams = streams;
	this.outputPath = path.resolve($tw.boot.wikiPath,$tw.config.wikiOutputSubDir);
};
```

With Prettier, it would look like the following:

```js
var Commander = function (commandTokens, callback, wiki, streams) {
  var path = require("path");
  this.commandTokens = commandTokens;
  this.nextToken = 0;
  this.callback = callback;
  this.wiki = wiki;
  this.streams = streams;
  this.outputPath = path.resolve(
    $tw.boot.wikiPath,
    $tw.config.wikiOutputSubDir
  );
};
```

**How can I help?**

If accepted, I would be willing to contribute by:
* setting up the repository to use Prettier
* fix linting rules is needed
* setting up Prettier to be run as part of CI jobs

**Comment by pmario** [comment]
You are right. There has been some discussion going on about "jslint" "header" settings lately. eg:

```
(function(){
/*jslint node: true, browser: true */                  <-- those 2 lines
/*global $tw: false */
"use strict";
```

To adjust the linter settings, it would be needed to amend a js code-guide first. 

I'm in favour of a general style-guide, but it needs to fit our requirements.

TW requirements are _not_ developer centric. We need to think about our "end-users", which is also true for the core js code. 

-----------

Just my thoughts. 

There is a little bit of reasoning, why TW code looks like, as it looks like: https://github.com/Jermolene/TiddlyWiki5/pull/4371#issuecomment-614546196

All TW modules use a `(function(){` wrapper. I think there is no need to add whitespace to the body code, or the wrapper itself. ..."prettier playground" seems to add whitespace here.

Is it possible to configure these setting?

I think prettier is nice, if a minification run is done afterwards. ... But that's not the case in TW. 

I think TW code is very close to https://github.com/airbnb/javascript/tree/es5-deprecated/es5 styleguide, with some exceptions. ... 

TW is ES5 only because of backwards and browser compatibility, so I think that's a good starting point.

Adding a 3rd party server as a dependency would be a no go for me. It needs to be a plugin for the editor and I intend to continue to use brackets.io 

I'm also not very happy with the 2 space indentation. If tabs are used, I can set them in the editor to be shown as 2,3,4 up to 8 spaces. Whatever I want. 

Loosing this possibility would be painful. 

Adding spaces instead of tabs and adding spaces after the `,` in function headers, will increase the core size by about 150kByte, or even more, which will directly increase empty.html by the same amount.

The primary usecase for TW still is the single file wiki. So initial size does matter and empty.html is already 2.2MByte in size. 

I think we should try to make the core smaller _and_ keep it readable. I personally would prefer to add 150kByte of code instead of whitespace.

**Comment by Jermolene** [comment]
Apologies for the late reply. I like the idea of having automated tests to enforce coding style. I've seen a few things like Prettier popup over the years (standard.js was popular for a long time too). My concerns are:

* That we choose Prettier and then find that the world moves on to another tool, and we have the costs of migration (this happened to us with Jasmine, arguably)
* That the initial switch over to a new coding style will be error prone and difficult to check. I think I'd want to use AST comparison tools to verify that nothing semantic has changed.

**Comment by pmario** [comment]
v5.4.0 may have a solution for this one using ESLint Styling
